### ISSUE TITLE
When validating a product with a bad schematron definition, bundle validation also fails indicating the associated product does not exist

### ISSUE THREAD

**Issue by jordanpadams** [NASA-PDS/validate#291]
**Describe the bug**
When I run validate against a bundle with a product that has invalid schematron definitions, it looks like validate is include that product in the bundle validation for referential integrity.

**To Reproduce**
1. when i run validate like this:

```
bin/validate -R pds4.bundle -t $TEST_DATA_PATH/test-bundles/pds4-compil-comet-v1.0/bundle.xml
```

2. one of the warning messages you get for product validation is:

```
  PASS: file:/data/local/starbase/data/pds4/test-bundles/pds4-compil-comet-v1.0/phys_char/document/pccref.xml
      WARNING  [warning.label.bad_schematypens]   Could not find expected pattern [schematypens=] in label file:/data/local/starbase/data/pds4/test-bundles/pds4-compil-comet-v1.0/phys_char/document/pccref.xml
      WARNING  [warning.label.missing_schematron_spec]   No schematrons specified in the label
        53 product validation(s) completed
```

3. And then later on during the bundle validation portion you get this erroneous warning:

```
  PASS: file:/data/local/starbase/data/pds4/test-bundles/pds4-compil-comet-v1.0/phys_char/document/pccref.txt
      WARNING  [warning.file.not_referenced_in_label]   File is not referenced by any label
        46 integrity check(s) completed
```

the label does in fact refer to this product:
```
            <Document_File>
                <file_name>pccref.txt</file_name>
                <document_standard_id>7-Bit ASCII Text</document_standard_id>
            </Document_File>
```

**Expected behavior**
if I fix the typo in the schematron

**Version of Software Used**
```
gov.nasa.pds:validate
Version 1.25.0-SNAPSHOT
Release Date: 2021-01-27 20:19:12

Copyright 2019, by the California Institute of Technology ("Caltech").
All rights reserved.
```
**Test Data / Additional context**
Can be found on pds-gamma under `$TEST_DATA_PATH/test-bundles/pds4-compil-comet-v1.0/`
